walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,115 @@
+            return Arrays.asList(TimeRange.WHOLE_DAY);
+        }
+
+        return available;
+    }
+
+    public List<TimeRange> remove(Collection<TimeRange> collectionRanges, TimeRange rangeToRemove) {
+        
+        int index = 0;
+        ArrayList<TimeRange> ranges = new ArrayList<>(collectionRanges);
+        
+        for (TimeRange range : ranges) {
+
+            // Case 1: range fully covers rangeToRemove
+            // |---------|       range
+            //   |----|          rangeToRemove
+            if (range.start() < rangeToRemove.start() && range.end() > rangeToRemove.end()) {
+                TimeRange leftRange = TimeRange.fromStartEnd(range.start(), rangeToRemove.start(), false);
+                TimeRange rightRange;
+                
+                if (range.end() >= TimeRange.END_OF_DAY) {
+                   rightRange = TimeRange.fromStartEnd(rangeToRemove.end(), TimeRange.END_OF_DAY, true);
+                } else {
+                    rightRange = TimeRange.fromStartEnd(rangeToRemove.end(), range.end(), true);
+                }
+                
+                ranges.set(index, leftRange);
+                ranges.add(index+1, rightRange);
+                
+                break;
+            }
+
+            // Case 2: range is fully contained in rangeToRemove
+            //   |---|            range
+            // |--------|         rangeToRemove
+            if (range.start() >= rangeToRemove.start() && range.end() <= rangeToRemove.end()) {
+                ranges.remove(index);
+                index--;
+                continue;
+            }
+
+            // Case 3: range starts before but end while rangeToRemove is happening
+            //   |------|         range
+            //      |--------|    rangeToRemove
+            if (range.start() < rangeToRemove.start() && range.end() >= rangeToRemove.start() && range.end() <= rangeToRemove.end()) {
+                TimeRange newRange = TimeRange.fromStartEnd(range.start(), rangeToRemove.start(), false);
+                ranges.set(index, newRange);
+                continue;
+            }
+
+            // Case 4: range ends after but starts while rangeToRemove is happening
+            //   |--------|       range
+            // |------|           rangeToRemove
+            if (range.start() >= rangeToRemove.start() && range.start() <= rangeToRemove.end() && range.end() > rangeToRemove.end()) {
+                TimeRange newRange = TimeRange.fromStartEnd(rangeToRemove.end(), range.end(), false);
+                ranges.set(index, newRange);
+                continue;
+            }
+
+            index++;
+        
+        }
+
+        return ranges;
+    }
+
+    public boolean attendeesOverlap(Set<String> setA, Set<String> setB) {
+        Set<String> intersection = new HashSet<>(setA);
+        intersection.retainAll(setB);
+        
+        if (intersection.isEmpty()) {
+            return false;
+        } else {
```

Comment:
We don't need explicit else clause here because we return in if, so it can be:
```
if (intersection.isEmpty()) {
    return false;
} 
return true;
```

Reply:
Oh right! Okay sure I'll make that change.